### DISCUSSION TITLE
Problem. How to? Docker + Fail2ban + Authelia 🤷🏻‍♂️ [SOLVED]

### DISCUSSION THREAD

**Discussion by taimadoCE** [authelia/authelia#4300] [Q&A] [answered]
Hi, friends. I'm trying to install Fail2ban into docker for monitoring the logs of the container of Authelia.
The main problem I found is that I don't know how to run with Authelia...

![imagen](https://user-images.githubusercontent.com/57089805/198874298-9296a528-1154-49b4-98b2-68a8497ec153.png)

I'm reading the section of logs in the web of Authelia.
![imagen](https://user-images.githubusercontent.com/57089805/198874414-197d6f6a-d933-49b8-8c52-1e89f6e2bbcf.png)

I get the log output as you can see:
![imagen](https://user-images.githubusercontent.com/57089805/198874445-19d8d731-54a8-4ec3-894e-977d4c1fd3c2.png)

My jail.local
![imagen](https://user-images.githubusercontent.com/57089805/198880858-ac42b2b6-81c5-4adb-8dee-16aeacfdfea1.png)

My configure.conf Authelia section of logs:
![imagen](https://user-images.githubusercontent.com/57089805/198880924-d5260fd8-806f-4b05-abf5-b7143c663aca.png)

The output of Fail2ban:
![imagen](https://user-images.githubusercontent.com/57089805/198881087-b7f4442f-42ff-4f16-8b4e-ca91943cfb6b.png)

And you can see up that the authelia.log exits in that folder...

Any idea?

**Comment by davama** [comment]
At least with the logs, you can mount a directory on the host itself which points to the authelia.log file.

```
-v /location/on/host/authelia.log:/log/authelia.log
```

Then mount that also on fail2ban container so it can keep track of it

Now what i would be curious about is if fail2ban can block traffic on the docker network (assuming fail2ban is also a container). Honestly dont know. Never tried that scenario

hope that helps

**Reply by taimadoCE** [reply]
Is that the path of logs folder in the container of Authelia?

**Comment by davama** [comment]
Define your logpath here:
https://github.com/authelia/authelia/blob/db9de7b5a4ad8fc8265fac4c25f9d798980db355/config.template.yml#L118

**Reply by taimadoCE** [reply]
I did that before
![imagen](https://user-images.githubusercontent.com/57089805/198895163-5e9f98ec-5cea-4eb7-9a77-537229250faf.png)
But, How can I do to read that file from the File2ban container? Always error archive not found, and as you can see the file is there...

**Comment by davama** [comment]
On the fail2ban container, did you mount the shared volume?

-v /location/on/host/authelia.log: /fail2ban/log/authelia.log

the above directory is being shared by both authelia and f2b. Authelia updates the file, f2b monitors it.

then configure your jail config with its `logpath = /fail2ban/log/authelia.log` as shown here: https://github.com/fail2ban/fail2ban/wiki/Fail2Ban-and-Docker

**Reply by taimadoCE** [reply]
Doesn't work.
![imagen](https://user-images.githubusercontent.com/57089805/198902459-1eb9d9d4-b8f1-463b-9511-33d313c05ea6.png)

It creates a folder, not an archive.
![imagen](https://user-images.githubusercontent.com/57089805/198902438-44416349-ca7b-4e96-8e39-ce7ab3e73ec7.png)

Int the portainer you can see this:
![imagen](https://user-images.githubusercontent.com/57089805/198902516-556c2a38-6917-4649-b58b-4c9f5efefe12.png)

**Comment by davama** [comment]
what about mounting the directory instead of the file.

-v /storage/.config/fail2ban/log:/authelia/log



-v /storage/.config/fail2ban/log:/fail2ban/log

**Comment by james-d-elliott** [comment, marked as answer]
You're mounting the wrong path. It should be `/storage/.config/authelia/config/authelia.log:/fail2ban/log/authelia.log:ro`. i.e. mounting the `/storage/.config/authelia/config/authelia.log` file into fail2ban container at the path `/fail2ban/log/authelia.log` readonly. 

`/storage/.config/fail2ban/log/authelia.log` isn't a file, it's a directory, probably created by docker when you tried to mount it.

**Reply by taimadoCE** [reply]
It works! Thanks again. I don't understand perfectly the syntax but Ill study it...Thanks.
